org.civicrm.reqangular/test/decorators/ui-bootstrap/uib-modal.decorator.spec.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'common/lodash',
+  'common/decorators/ui-bootstrap/uib-modal.decorator',
+  'common/angularMocks',
+  'common/angular',
+  'common/angularBootstrap'
+], function (_, $uibModalDecorator) {
+  'use strict';
+
+  describe('$uibModal.open', function () {
+    var $uibModal, $document;
+
+    beforeEach(module('ui.bootstrap'));
+
+    beforeEach(module(function ($provide) {
+      $provide.decorator('$uibModal', $uibModalDecorator);
+    }));
+
+    beforeEach(inject(function (_$document_, _$uibModal_) {
+      $document = _$document_;
+      $uibModal = _$uibModal_;
+    }));
+
+    describe('init', function () {
+      var originalStyles;
+      var arg1 = { template: '<component></component>' };
+      var arg2 = 'another_argument';
+      var elements = {};
+
+      beforeEach(function () {
+        elements.body = $document[0].body;
+        elements.html = $document[0].getElementsByTagName('html')[0];
+
+        if (!originalStyles) {
```

Comment:
Why do we need to check this condition?

Reply:
I need to assign in only once. I tried beforeAll, but it is executed after nested beforeEach.